test/fuzz/lua_fuzzer/serializer.cc

```diff
@@ -0,0 +1,608 @@
+	return ret;
+}
+
+PROTO_TOSTRING(TableConstructor, table)
+{
+	std::string ret = "{ ";
+	if (table.has_list())
+		ret += FieldListToString(table.list());
+	ret += " }";
+	return ret;
+}
+
+PROTO_TOSTRING(FieldList, list)
+{
+	std::string ret = FieldToString(list.firstfield());
+	for (int i = 0; i < list.fields_size(); ++i)
+		ret += FieldWithFieldSepToString(list.fields(i));
+	if (list.has_lastsep())
+		ret += FieldSepToString(list.lastsep());
+	return ret;
+}
+
+PROTO_TOSTRING(FieldWithFieldSep, field)
+{
+	std::string ret = FieldSepToString(field.sep());
+	ret += " ";
+	ret += FieldToString(field.cur_field());
+	return ret;
+}
+
+PROTO_TOSTRING(Field, field)
+{
+	using FieldType = Field::FieldOneofCase;
+	switch (field.field_oneof_case()) {
+	case FieldType::kExprDouble:
+		return ExpressionEqualExpressionToString(field.exprdouble());
+	case FieldType::kName:
+		return NameEqualExpressionToString(field.name());
+	case FieldType::kExpr:
+		return ExpressionToString(field.expr());
+	default:
+		/* More common case of using fields. */
+		return NameEqualExpressionToString(field.name());
+	}
+}
+
+PROTO_TOSTRING(ExpressionEqualExpression, equality)
+{
+	std::string ret = "[ ";
+	ret += ExpressionToString(equality.leftexpr());
+	ret += " ]";
+	ret += " = ";
+	ret += ExpressionToString(equality.rightexpr());
+	return ret;
+}
+
+PROTO_TOSTRING(NameEqualExpression, equality)
+{
+	std::string ret = ConvertToString(equality.name());
+	ret += " = ";
+	ret += ExpressionToString(equality.expr());
+	return ret;
+}
+
+PROTO_TOSTRING(FieldSep, sep)
+{
+	using FieldSepType = FieldSep::SepOneofCase;
+	switch (sep.sep_oneof_case()) {
+	case FieldSepType::kComma:
+		return ",";
+	case FieldSepType::kSemicolon:
+		return ";";
+	default:
+		return ",";
+	}
+}
+
+PROTO_TOSTRING(BinaryOperator, op)
+{
+	using BinopType = BinaryOperator::BinaryOneofCase;
+	switch (op.binary_oneof_case()) {
+	case BinopType::kAdd:
+		return "+";
+	case BinopType::kSub:
+		return "-";
+	case BinopType::kMult:
+		return "*";
+	case BinopType::kDiv:
+		return "/";
+	case BinopType::kExp:
+		return "^";
+	case BinopType::kMod:
+		return "%";
+	case BinopType::kConcat:
+		return "..";
+	case BinopType::kLess:
+		return "<";
+	case BinopType::kLessEqual:
+		return "<=";
+	case BinopType::kGreater:
+		return ">";
+	case BinopType::kGreaterEqual:
+		return ">=";
+	case BinopType::kEqual:
+		return "==";
+	case BinopType::kNotEqual:
+		return "~=";
+	case BinopType::kAnd:
+		return "and";
+	case BinopType::kOr:
+		return "or";
+	default:
+		/* Works in most cases. */
+		return "==";
+	}
+}
+
+PROTO_TOSTRING(UnaryOperator, op)
+{
+	using UnaryopType = UnaryOperator::UnaryOneofCase;
+	switch (op.unary_oneof_case()) {
+	case UnaryopType::kNegate:
+		return "-";
+	case UnaryopType::kNot:
+		return "not ";
+	case UnaryopType::kLength:
+		return "#";
+	default:
+		/* Works in most cases. */
+		return "not ";
+	}
+}
+
+std::string
+ConvertToString(const std::string &s, const std::string &def)
+{
```

Comment:
I think that is because in protobuf all string fields in messages are empty by default, and (I don't know why) they rarely mutate, so in ConvertToString function string 's' is empty in most cases and thus function is returning default value.

Reply:
Can we use `mutator::MutateString()` directly for string mutations, unless we get well-suited string? See some examples [here](https://github.com/google/libprotobuf-mutator/blob/dbe588bfb6922060e557fe5b8ee27d2923000c1a/src/mutator.h#L34). Or it is useless, because we must use the same mutator for the whole code generation? @ligurio, what do you think?


Also, can we separate name generation and string generations to avoid names like in the example above (for example started with number)?
```lua
function a.5555555555555555555.5555555555555555555 ( a ) end
```

Reply:
Added logic for creation more lua identifiers like "Name" + some number % kMaxIdentifiers.